meson.build

```diff
@@ -55,6 +55,14 @@ add_project_arguments('-Wno-reorder-init-list', language : ['cpp'])
 add_project_arguments('-Wno-unknown-warning-option', language : ['cpp'])
 add_project_arguments('-Wno-unused-parameter', language : ['c', 'cpp'])
 
+if get_option('buildtype') == 'release'
+  add_project_arguments('-O3',  language : ['c', 'cpp'])
+endif
+
+if get_option('buildtype') == 'debugoptimized'
+  add_project_arguments('-O3', '-g3',  language : ['c', 'cpp'])
+endif
```

Comment:
Actually I would say that this is all wrong. The builtin "buildtype" option is specifically supposed to handle this for you:
https://github.com/xmake-io/xmake-repo/pull/1837#issuecomment-1432510233

For release it already adds -O3.
For debugoptimized it already adds `-g2 -O2`, which is usually considered a pretty decent balance...